src/api/lib/storage/backends/consul.js

```diff
@@ -0,0 +1,109 @@
+var Storage = require('../');
+var consul = require('consul');
+var path = require('path');
+
+module.exports = ConsulBackend;
+
+function ConsulBackend(options) {
+	options = options || {};
+	this._prefix = options.prefix || 'billimanjaro';
+	delete options.prefix;
+	this._instance = consul(options);
+}
+
+ConsulBackend.prototype = Object.create(Storage.prototype);
+ConsulBackend.prototype.constructor = Storage;
+ConsulBackend.prototype._idGenerator = function() { return Math.random().toString(36).slice(2); };
+
+function syncEntity(backend, entity, id, object, callback) {
+	if(Object.keys(object).length === 0 ||
+		(Object.keys(object).length === 1 &&
+		Object.keys(object)[0] === 'id')) {
```

Comment:
Shouldn't you just check if `id` is contained rather than asserting the location in the array?